executor/predictor/metadata_test.go

```diff
@@ -0,0 +1,199 @@
+package predictor
```

Comment:
Writing tests for cases where `nil, nil` is returned as `input` and `output` may help confirm if it is correct logic in the cases mentioned above.

Reply:
See comment above about the corner cases.